src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -109,12 +140,21 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
 }
 
 object Iterator {
+
   val empty: Iterator[Nothing] = new Iterator[Nothing] {
     def hasNext = false
     def next() = throw new NoSuchElementException("next on empty iterator")
   }
+
+  def single[A](a: A): Iterator[A] = new Iterator[A] {
+    private var consumed: Boolean = false
+    def hasNext = !consumed
+    def next() = if (consumed) empty.next() else { consumed = true; a }
+  }
+
   def apply[A](xs: A*): Iterator[A] = new IndexedView[A] {
```

Comment:
This is going to keep the pointless box of `IndexedView` around, isn't it?  This should be as thin a wrapper around `xs` as possible.  (Maybe this PR isn't the place to fix it, but if you're adding `single`, why not get this also?)